src/router/controller/params/decorators.ts

```diff
@@ -37,13 +37,22 @@ export const Res = createParamDecorator((context: HttpContext) => {
 });
 
 export const Body = (prop?: string) =>
-	createParamDecorator((context: HttpContext) => {
-		if (prop) {
-			// deno-lint-ignore no-explicit-any
-			return (context.request.body as any)[prop];
-		} else {
-			return context.request.body;
+	createParamDecorator(async (context: HttpContext) => {
+		let body;
+		try {
+			if (context.request.body instanceof Function) {
+				body = await context.request.body()?.value;
+			} else {
+				body = await context.request.body;
+			}
+		} catch {
+			return null;
+		}
+
+		if (!body) {
+			return null;
 		}
+		return prop ? body[prop] : body;
```

Comment:
What behavior requires these changes? I do not see any new tests

Reply:
The tests are not correctly covering this area as the force an object but in practice body is a function.

Try the following from the original code:
1. Create a POST route with a `@Body() body: any` that simply return the body sent
2. Send a request with curl
3. Try again with my PR 😛

Reply:
Yeah but you agree we won't manually do that every time we change something in the codebase 😛 so we need to add tests